javascript/modules/client/src/client.ts

```diff
@@ -366,6 +372,14 @@ export class Client extends ClientCore implements IClient {
       signer,
     );
 
+    // @ts-ignore
+    let cid: string;
+    try {
+      cid = await this.ipfs.add(JSON.stringify(params.metadata));
+    } catch {
+      throw new Error("Could not pin the metadata on IPFS");
```

Comment:
We might create and use custom errors in sdk/common

Reply:
Yes I've been thinking on this, but would create a separate task to change all the error messages to the errors in common

Reply:
Well, if the PR was unrelated to this feature... it might make sense, but if we're adding the functionality now, I don't see why adding technical debt would be a gain :)